TFUtil.py

```diff
@@ -1249,6 +1249,50 @@ def copy_template(self, name=None, dtype=None):
       kwargs["dtype"] = dtype
     return Data(**kwargs)
 
+  def copy_template_excluding_axis(self, exclude_axis, name=None):
+    """
+    :param int exclude_axis: axis to be removed.
+    :param str|None name: if set, this will be the new name.
+    :return: copy of myself excluding remove_axis_num axis, without placeholder.
+    :rtype: Data
+    """
+    kwargs = self.get_kwargs()
+    if exclude_axis < 0:
+      exclude_axis += self.batch_ndim
+      assert exclude_axis >= 0
+    assert 0 <= exclude_axis < self.batch_ndim
+    axis_to_exclude_wo_b = self.get_batch_axis_excluding_batch(exclude_axis)  # None if exclude_axis == batch_dim_axis.
+    if exclude_axis == self.feature_dim_axis:
+      del kwargs["dim"]
+
+    other_special_axes = self.get_special_axes_dict(
+      counted_with_batch_dim=True, only_available=True, include_batch_dim_axis=True)
+    for axis_name, axis in other_special_axes.items():
+      if axis == exclude_axis and axis_name in kwargs:
```

Comment:
Ah sorry, `axis_name in kwargs` does not make sense here (if you have that `else:` part there), because in the `else:` part, you assume that it is always in `kwargs` (otherwise you should not touch it).
As you have `only_available=True` above, I guess it is anyway always already in. You can just check by:
`assert axis_name in kwargs` (before the `if`)